docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,529 @@
+* Bad, because user can forget about it. 
+User assumes MeshHTTPRoute affects all consumers while in reality it affects only consumers in the same zone.
+
+##### Pros and Cons of introducing `spec.export.mode`, the default is `All`
+
+* Good, because coherent with MeshService.
+* Good, because treats all consumers equally by default without favoring the local ones.
+* Bad, because it's not backwards compatible.
+* Bad, because it doesn't add much value.
+It can be easily replaced with the top-level `targetRef{kind:MeshSubset,tags{"kuma.io/zone":"my-zone"}}` when user wants to opt-out.
+
+##### Pros and Cons of automatically detecting if a policy needs to be exported
+
+Having `spec.export.mode` might be excessive taking into account we already have a top-level targetRef.
+If the top-level targetRef specifies a tag `kuma.io/zone` then it's clear whether we need to export the policy or not.
+
+Zone-originated policies in `kuma-system` won't be synced to other zones. 
+This eliminated the problem of backwards compatibility. 
+Also, with the introduction of namespace-originated policies, applying policies to `kuma-system` becomes an edge use case.
+The only reason to apply a policy on zone's `kuma-system` is Zone Cluster Operator stepping in to fix the undesired behaviour.
+
+Zone-originated policies on Universal won't be synced to other zones as well. 
+We might want to introduce namespaces to Universal in the future.
+
+* Good, because we don't introduce new fields, no need for users to learn new concepts.
+* Good, because all consumers are treated equally by default without favoring the local ones.
+* Bad, because if you want to keep this only to your local zone, you need to specify the name of the zone in the policy
+
+### Order when merging
+
+By design, several policies can select the same DPP.
+In that case, policies are ordered and their `to` (or `from`) lists are merged.
+Introducing namespace-scoped policies adds another steps to the policy comparison process:
+
+1. Is top-level `targetRef.kind` more specific (MeshServiceSubset > MeshService > MeshSubset > Mesh)?
+   When kinds are equal, go to the next step.
+2. Zone originated policy is more specific than global originated policy.
+   When policies originates at the same place, go to the next step.
+3. **[new step]** Policy in any custom namespace is more specific than any policy in the `kuma-system` namespace.
+   When both policies either from custom namespaces or from `kuma-system`, go to the next step.
+4. **[new step]** Policy from local zone is more specific than a policy from another zone. 
+   When policies have the same zone, go to the next step.
+5. **[new step]** Policy from local namespace is more specific than a policy from another namespace.
+   When namespaces are equal, go to the next step.
+6. The policy is more specific if it's name is lexicographically less than other policy's name ("aaa" < "bbb" so that "
```

Comment:
Can we just write pseudo code? This is hard to process

Reply:
We tried, it doesn't look simpler, we merged 3 steps into 1 so should be easier to process